designer/server/src/main/scala/pl/touk/nussknacker/ui/suggester/ExpressionSuggester.scala

```diff
@@ -30,6 +30,25 @@ class ExpressionSuggester(
       scenarioPropertiesNames
     )
 
+  private def truncateExpressionByCaretPosition2d(
```

Comment:
All heuristics should be in internals (implementation class) as it is a part of the code that would change frequently. Also it would be nice to keep all heuristics somehow organized to help diagnosis what happen in some surprising situations - see `shouldInsertDummyVariable`